Stuck on a particular block while syncing: Internal error while importing block

I'm trying a fresh sync on Medalla, but have become stuck on a particular block. The following message just repeats exactly (with the same block root mentioned) every few seconds and the sync is not making progress. Note that this block is from long before the current network chaos.

```
08:45:30.362 WARN  - Failed to import block from peer (err: INTERNAL_ERROR) SignedBeaconBlock{message=BeaconBlock{root=0x9054b9a5ee54075f3b1251f2892b4df7034612be384c0dfc245e6ac93c98b613, slot=39425, proposer_index=>
08:45:32.375 INFO  - Sync Event  *** Current slot: 77927, Head slot: 39423, Connected peers: 80
08:45:39.342 ERROR - Internal error while importing block: BeaconBlock{root=0x9054b9a5ee54075f3b1251f2892b4df7034612be384c0dfc245e6ac93c98b613, slot=39425, proposer_index=18074, parent_root=0xb351150d7786a65fe38c9c>
java.util.concurrent.CompletionException: java.lang.IllegalArgumentException: State must belong to the given block
        at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331) ~[?:?]
        at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346) ~[?:?]
        at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:704) ~[?:?]
        at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
        at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2159) ~[?:?]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateResult$2(SafeFuture.java:129) ~[teku-infrastructure-async-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859) ~[?:?]
        at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837) ~[?:?]
        at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
        at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2159) ~[?:?]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateToAsync$11(SafeFuture.java:222) ~[teku-infrastructure-async-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.fromRunnable(SafeFuture.java:138) ~[teku-infrastructure-async-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        at tech.pegasys.teku.infrastructure.async.AsyncRunner.lambda$runAsync$0(AsyncRunner.java:24) ~[teku-infrastructure-async-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.ofComposed(SafeFuture.java:85) ~[teku-infrastructure-async-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.lambda$createRunnableForAction$1(ScheduledExecutorAsyncRunner.java:119) ~[teku-infrastructure-async-0.12.4-SNAPSHOT.jar:0.12.4-dev-df30>
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:830) [?:?]
Caused by: java.lang.IllegalArgumentException: State must belong to the given block
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:142) ~[guava-29.0-jre.jar:?]
        at tech.pegasys.teku.datastructures.blocks.SignedBlockAndState.<init>(SignedBlockAndState.java:32) ~[teku-ethereum-datastructures-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        at tech.pegasys.teku.storage.server.rocksdb.RocksDbDatabase.getFinalizedBlockAndState(RocksDbDatabase.java:446) ~[teku-storage-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        at tech.pegasys.teku.storage.server.rocksdb.RocksDbDatabase.updateFinalizedData(RocksDbDatabase.java:402) ~[teku-storage-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        at tech.pegasys.teku.storage.server.rocksdb.RocksDbDatabase.doUpdate(RocksDbDatabase.java:350) ~[teku-storage-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        at tech.pegasys.teku.storage.server.rocksdb.RocksDbDatabase.update(RocksDbDatabase.java:173) ~[teku-storage-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        at tech.pegasys.teku.storage.server.ChainStorage.lambda$onStorageUpdate$0(ChainStorage.java:92) ~[teku-storage-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.fromRunnable(SafeFuture.java:138) ~[teku-infrastructure-async-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        at tech.pegasys.teku.storage.server.ChainStorage.onStorageUpdate(ChainStorage.java:90) ~[teku-storage-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        at jdk.internal.reflect.GeneratedMethodAccessor31.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:567) ~[?:?]
        at tech.pegasys.teku.events.DirectEventDeliverer.executeMethod(DirectEventDeliverer.java:74) ~[teku-events-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        at tech.pegasys.teku.events.DirectEventDeliverer.deliverToWithResponse(DirectEventDeliverer.java:67) ~[teku-events-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        at tech.pegasys.teku.events.AsyncEventDeliverer.lambda$deliverToWithResponse$1(AsyncEventDeliverer.java:75) ~[teku-events-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        at tech.pegasys.teku.events.AsyncEventDeliverer$QueueReader.deliverNextEvent(AsyncEventDeliverer.java:117) ~[teku-events-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        at tech.pegasys.teku.events.AsyncEventDeliverer$QueueReader.run(AsyncEventDeliverer.java:109) ~[teku-events-0.12.4-SNAPSHOT.jar:0.12.4-dev-df300e3b]
        ... 3 more
08:45:39.349 WARN  - Failed to import block from peer (err: INTERNAL_ERROR) SignedBeaconBlock{message=BeaconBlock{root=0x9054b9a5ee54075f3b1251f2892b4df7034612be384c0dfc245e6ac93c98b613, slot=39425, proposer_index=>
08:45:44.400 INFO  - Sync Event  *** Current slot: 77928, Head slot: 39423, Connected peers: 79
08:45:47.597 ERROR - Internal error while importing block: BeaconBlock{root=0x9054b9a5ee54075f3b1251f2892b4df7034612be384c0dfc245e6ac93c98b613, slot=39425, proposer_index=18074, parent_root=0xb351150d7786a65fe38c9c>
java.util.concurrent.CompletionException: java.lang.IllegalArgumentException: State must belong to the given block
```

Version: teku/v0.12.4-dev-df300e3b/linux-x86_64/-privatebuild-openjdk64bitservervm-java-13

--- a/storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java
+++ b/storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java
@@ -492,7 +492,7 @@ public class RocksDbDatabase implements Database {
       switch (stateStorageMode) {
         case ARCHIVE:
           // Get previously finalized block to build on top of
-          final SignedBlockAndState baseBlock = getFinalizedBlockAndState();
+          final SignedBeaconBlock baseBlock = getFinalizedBlock();
 
           final HashTree blockTree =
               HashTree.builder()
@@ -541,11 +541,9 @@ public class RocksDbDatabase implements Database {
     }
   }
 
-  private SignedBlockAndState getFinalizedBlockAndState() {
+  private SignedBeaconBlock getFinalizedBlock() {
     final Bytes32 baseBlockRoot = hotDao.getFinalizedCheckpoint().orElseThrow().getRoot();
-    final SignedBeaconBlock baseBlock = finalizedDao.getFinalizedBlock(baseBlockRoot).orElseThrow();
-    final BeaconState baseState = hotDao.getLatestFinalizedState().orElseThrow();
-    return new SignedBlockAndState(baseBlock, baseState);
+    return finalizedDao.getFinalizedBlock(baseBlockRoot).orElseThrow();
   }
 
   private void putFinalizedState(

